stm32-modules/heater-shaker/simulator/sim_driver.cpp

```diff
@@ -0,0 +1,101 @@
+#include "simulator/sim_driver.hpp"
+
+#include <boost/asio.hpp>
+#include <regex>
+
+#include "simulator/simulator_queue.hpp"
+
+using namespace sim_driver;
+
+sim_driver::SocketSimDriver::SocketSimDriver(std::string url) {
+    std::regex url_regex(":\\/\\/([a-zA-Z0-9.]*):(\\d*)$");
+    std::smatch url_match_result;
+
+    if (std::regex_search(url, url_match_result, url_regex)) {
+        host = url_match_result[1];
+        port = std::stoi(url_match_result[2]);
+
+    } else {
+        std::cerr << "Malformed url." << std::endl;
+        exit(1);
+    }
+}
+
+std::string sim_driver::SocketSimDriver::get_host() { return this->host; }
+
+int sim_driver::SocketSimDriver::get_port() { return this->port; }
+
+std::string sim_driver::SocketSimDriver::get_name() { return this->name; }
+
+void sim_driver::SocketSimDriver::write() {}
+
+auto get_socket(std::string host, int port) {
+    boost::asio::io_service io_context;
```

Comment:
If you made the `Socket` class discussed above actually own the _boost asio socket_, you could wrap all this up with similar initialization/destruction guarantees.